funflow-tutorial/notebooks/CustomTaskAndConfigurable.ipynb

```diff
@@ -0,0 +1,310 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    ":opt no-lint\n",
+    "-- Required language extensions\n",
+    "{-# LANGUAGE DataKinds #-}             -- typelevel lists and tuples\n",
+    "{-# LANGUAGE GADTs #-}                 -- task declaration\n",
+    "{-# LANGUAGE OverloadedLabels #-}      -- named task disambiguation\n",
+    "{-# LANGUAGE OverloadedStrings #-}     -- flexible string types\n",
+    "{-# LANGUAGE RankNTypes #-}            -- polymorphism of Flow\n",
+    "\n",
+    "-- core types and functions used here\n",
+    "import Funflow.Config (render, Configurable (..), ExternalConfig)\n",
+    "\n",
+    "-- other necessary imports\n",
+    "import System.Environment (setEnv, unsetEnv)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Custom tasks, `Configurable`s, and type checking\n",
+    "\n",
+    "In this tutorial, we'll show how to build a `Flow` that uses as custom task type.\n",
+    "Additionally, we'll demonstrate `funflow`'s type safety for `Configurable` parameters."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining a custom task type and its _interpreter_\n",
+    "\n",
+    "Here, we declare a custom task type and then define __what it does__ with an _interpreter_ function."
```

Comment:
I think "how it is executed" or something similar might be a bit clearer here since the tasks should hold some intrinsic meaning (e.g. the task is "run a command in a docker container" and the interpreter picks which executable to use, how to handle errors, etc.)